crates/swc_ecma_transforms_compat/src/es2020/optional_chaining.rs

```diff
@@ -64,6 +64,7 @@ impl VisitMut for OptChaining {
             Expr::OptChain(v) => {
                 let data = self.gather(v.take(), vec![]);
                 *e = self.construct(data, false, self.c.no_document_all);
+                e.visit_mut_children_with(self);
```

Comment:
We would need the same call for `Expr::Unary`, but I think the better solution is to fix it at https://github.com/kdy1/swc/blob/9f44d10bb2a28aa7d776c2a4243074e8f59494f7/crates/swc_ecma_transforms_compat/src/es2020/optional_chaining.rs#L207. That'll avoid a repeated iteration of `prop` and `args` (we do them during `gathering`). The only thing I missed was transforming the `base`